Modules/unicodedata.c

```diff
@@ -1477,13 +1491,19 @@ unicodedata_exec(PyObject *module)
     }
 
     /* Export C API */
-    v = PyCapsule_New((void *)&unicodedata_capi, PyUnicodeData_CAPSULE_NAME,
-                      NULL);
-    if (v == NULL) {
+    _PyUnicode_Name_CAPI *capi = udc_get_capi();
+    if (capi == NULL) {
         return -1;
     }
-    if (PyModule_AddObject(module, "_ucnhash_CAPI", v) < 0) {
-        Py_DECREF(v);
+    PyObject *capsule = PyCapsule_New(capi,
+                                      PyUnicodeData_CAPSULE_NAME,
+                                      udc_destroy_capi);
+    if (capsule == NULL) {
+        PyMem_Free(capi);
+        return -1;
+    }
+    if (PyModule_AddObject(module, "_ucnhash_CAPI", capsule) < 0) {
```

Comment:
How about using `PyModule_AddObjectRef` in here too?

Reply:
We could do that, but I'm not sure it would improve this code. What do you think?

Reply:
`PyModule_AddObjectRef()` is more easy to understand than `PyModule_AddObject()`, because `PyModule_AddObject()` will steal the refs sometimes.

Reply:
After thinking about it, I do agree. It's easier to follow the ref count when reading the code. I'll change it. Thanks!

Reply:
Should I change the other `PyModule_AddObject` in `unicodedata_exec()` while we're there?